apis/projectcontour/v1/detailedconditions.go

```diff
@@ -114,3 +114,48 @@ type DetailedCondition struct {
 }
 
 const ValidConditionType string = "Valid"
+
+type ErrorType uint
+type WarningType uint
+
+func (ec ErrorType) String() string {
+	return errorConditions[ec]
+}
+
+func (wt WarningType) String() string {
+	return warningConditions[wt]
+}
+
+const (
+	IncludeError ErrorType = iota
+	AuthError
+	TCPProxyError
+	CORSError
+	RouteError
+	PrefixReplaceError
+	ServiceError
+	TCPProxyIncludeError
+	VirtualHostError
+	TLSError
+	RootNamespaceError
+	SpecError
+	OrphanedError
+)
+
+var errorConditions = []string{
```

Comment:
A follow-up would be to add docs to this, would be neat to autogenerate those types.